code/modules/recycling/conveyor2.dm

```diff
@@ -184,15 +211,24 @@
 	speed_process = 1
 
 
-/obj/machinery/conveyor_switch/New()
-	..()
+/obj/machinery/conveyor_switch/New(newloc, newid)
+	..(newloc)
+	if(!id)
+		id = newid
 	update()
 
-	spawn(5)		// allow map load
+/*	spawn(5)		// allow map load
 		conveyors = list()
-		for(var/obj/machinery/conveyor/C in world)
+		for(var/obj/machinery/conveyor/C in machines) //This works if it's: `C in world` , but that's bad practice. Best to try to fix it.
 			if(C.id == id)
-				conveyors += C
+				conveyors += C */ //Moving this to initialise to see if it gets around the bug... do we need the spawn thing? Let's find out.
+
+/obj/machinery/conveyor_switch/initialize() //This whole thing is new
+	conveyors = list()
+	for(var/obj/machinery/conveyor/C in machines)

```

Comment:
This will still not work. `machines += src` should be in `/obj/machinery/New`, and remove that line from `AddAtProcessing`